.storybook/preview.js

```diff
@@ -16,6 +16,58 @@ addDecorator(withA11y);
 const themes = [{ name: 'Dark', class: 't-dark', color: colors.primaryBrand }];
 addParameters({ themes });
 
+/**
+ * Define the type of value passed to `addParameters.options.storySort`, which is
+ * parsed to get to the name of each story's category
+ * @typedef { [String, import('@storybook/client-api/dist/types').StoreItem] } StoryItem
+ */
```

Comment:
Can you please include a link to where you found this information?

Reply:
TBH, I searched for the shape that matched what I was logging when testing how to get the value I needed for sorting. This led me to the `StoreItem` interface: https://github.com/search?q=StoreItem+in:file+extension:ts+repo:storybookjs/storybook&type=Code

To get the type back, I needed to reference the `/dist` rather than the `/src`.

I realize this project is not set up to use TypeScript by default, as Decathlon is. If this is too much magic for you, I can revert the use of the typings altogether.

This is the reference that shows the way to get at the values for an example sort: https://storybook.js.org/docs/configurations/options-parameter/#sorting-stories

Reply:
For context, this code example just made me think "Can I get some explanation of this magic?":

```js
storySort: (a, b) =>
      a[1].kind === b[1].kind
...
```

Reply:
Yeah, if this is a typescript definition that this information isn't relevant to this project.

Reply:
Then I'll revert this, as well. Understand that you can use TypeScript with JSDoc and plain JS files, but consuming type definitions. This is what we ended up doing elsewhere, but also had things set up to verify typings.

Reply:
FWIW I think these jsdoc annotations are useful, they are being used to power vscode's autocomplete and for me they are useful to help understand the code

Reply:
Thanks, @calebeby.

Trying to re-phrase so I'm clear:

It's fine to include `@typedef`, but for those of us unfamiliar with TypeScript it would help a lot to have some sort of `@see` link to a relevant URL, even if it is just the source file on GitHub where the type is defined.

Reply:
If you ctrl-click on `StoreItem` on that line in VSCode it will go to where it is defined in `node_modules`

Reply:
OK, I hope this update isn't going to spawn much more discussion. I went down the road of adding typings because I wanted to use JSDoc annotations. If that's still preferred, I can define a type and just describe the shape of the value used for sorting (`object` with a `kind`:`string` property). As it is, I ended up just describing what's going on and removed the JSDoc annotations. Added a reference to the `storySort` example in Storybook docs.

Reply:
My dudes: You're killing me here. As the person primarily responsible for maintaining this file, I'm asking for context. Not explanations of what `@typedef` is. Not instructions on how to use my editor. Concise, clear context for why the code is written.

Ideally, this would be in the comment itself. For example:

```
/**
 * The parameters for Storybook sorting functions are sparsely documented.
 * Including a detailed type definition in lieu of clear documentation to
 * link to.
 *
 * @typedef (...)
 */
```

For some reason, this feedback is not being addressed. So I've asked for a URL for context instead. Now I'm hearing that this isn't necessary because some of the team is aware of editor functions in a specific editor.

How are we still spinning on this?

Reply:
> If you ctrl-click on StoreItem on that line in VSCode it will go to where it is defined in node_modules

I took @tylersticka's question to mean "How did you find this," and the truth was, I did some digging. I may not have known the appropriate way to easily get to the interface definition, or it may be that something's not quite right with Storybook's typings, not sure.

I fear I may have gone too far in the other direction with my last update. I can revert that commit and revise as desired, if that would be better.

Sorry for the churn. I'm trying to provide decent documentation and not add confusion.

Reply:
> I'm sorry for sounding so exasperated about this, but I'm at a loss for why we're still spinning on such a small request.

I'm sorry, as well. If adding the comment you outlined above to the code as I had it in https://github.com/cloudfour/cloudfour.com-patterns/pull/607/commits/c3fa4d9783e501747975bc980c0a11d852267023, I will do that.

Reply:
@tylersticka, I have reverted back to the previous state, added the description on the `@typedef` you suggested, and added links to the GitHub source files for the interfaces that make up `StoreItem`. I hope this provides what you're looking for.